nx/lib/nx.ex

```diff
@@ -3265,6 +3265,129 @@ defmodule Nx do
     end)
   end
 
+  @doc ~S"""
+  Split a tensor into train and test subsets.
+
+  `split` split must be an integer greater than zero and less than the length of the tensor.
+
+  ## Options
+
+    * `:axis` - The axis along which to take the values from. Defaults to `0`.
+
+  ## Examples
+
+    Split a tensor into two separate tensors.
+
+    iex> {train, test} = Nx.split(Nx.tensor([[3, 6, 5], [26, 75, 3], [23, 4, 1]]), 2, axis: 0)
```

Comment:
Let's use the first example without passing the axis option.

I think an iota as the input tensor is more easily readable due to the inherent ordering as well.
And we could even have just a rank 1 tensor for the first example